tufaceous-lib/src/repository.rs

```diff
@@ -276,15 +278,13 @@ impl OmicronRepoEditor {
             artifact.target = filename.clone();
         } else {
             // if we don't, make sure we're not overriding another target.
-            if self.existing_targets.iter().any(|target_name| {
-                target_name.raw() == filename
-                    && target_name.resolved() == filename
-            }) {
+            if !self.existing_target_filenames.insert(filename.clone()) {
```

Comment:
It seems strange to me that we have to do this check ourselves to avoid clobbering another file, even if the metadata (its `kind`) and contents (including the hash) are different. Are we (ab)using tuf in some way that makes this necessary?